programs/drift/src/controller/orders.rs

```diff
@@ -1102,6 +1102,17 @@ pub fn fill_perp_order(
         amm_is_available &= !market.is_operation_paused(PerpOperation::AmmFill);
         amm_is_available &= !market.has_too_much_drawdown()?;
 
+        // We are already using safe oracle data from MM oracle.
+        // But AMM isnt available if we could have used MM oracle but fell back due to price diff
+        let amm_available_mm_oracle_recent_but_volatile =
+            if mm_oracle_price_data.is_mm_oracle_as_recent() && market.amm.mm_oracle_slot != 0 {
```

Comment:
nit: feel like i'd move the != 0 check in the the is_mm_oracle_as_recent change or id add a separate method `is_enabled`.